docs/TypeConstructor.md

```diff
@@ -6,75 +6,98 @@ value.
-new instance to be set.
+  const Example = DefineMap.extend({
+    prop: {
+      Type: Person
+    }
+  });
+  const ex = new Example();
+  ex.set( "prop", {first: "Justin", last: "Meyer"} );
 
-@signature `{propDefinition}`
+  console.log( ex.prop instanceof Person ); //-> true
+  console.log( ex.prop.fullName ); //-> "Justin Meyer"
+  ```
+  @codepen
 
-A [can-define.types.propDefinition] that defines an inline [can-define/map/map] type.  For example:
+  The `Type` constructor is called before the [can-define.types.type] property and before [can-define.types.set]. It checks if the incoming value
+  is an [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof) `Type`. If it is, or if it is `null` or `undefined`, it passes the original value through.  If not, it passes the original value to `new Type(originalValue)` and returns the
+  new instance to be set.
 
-```js
-{
-	address: {
-		Type: {
-			street: "string",
-			city: "string"
-		}
-	}
-}
-```
+@signature `{propDefinition}`
 
-@signature `[Type|propDefinition]`
+  A [can-define.types.propDefinition] that defines an inline [can-define/map/map] type.  For example:
 
-Defines an inline [can-define/list/list] type that's an array of `Type` or inline [can-define.types.propDefinition] [can-define/map/map]
-instances.  For example:
+  ```js
+  {
+    address: {
+      Type: {
+        street: "string",
+        city: "string"
+      }
+    }
+  }
+  ```
 
-```js
-{
-	people: {
-		Type: [ Person ]
-	},
-	addresses: {
-		Type: [ {
-			street: "string",
-			city: "string"
-		} ]
-	}
-}
-```
+@signature `[Type|propDefinition]`
 
+  Defines an inline [can-define/list/list] type that's an array of `Type` or inline [can-define.types.propDefinition] [can-define/map/map]
+  instances.  For example:
+
+  ```js
+  import {DefineMap} from "can";
+
+  const List = DefineMap.extend({
+    people: {
+      Type: [ Person ]
+    },
+    addresses: {
+      Type: [ {
+        street: "string",
+        city: "string"
+      } ]
+    }
+  });
+
+  const myList = new List({
+    people: [ {first: "Justin", last: "Meyer"} ],
+    addresses: [ {street: "11 Example Ave.", city: "Chicago"} ]
+  });
+
+  console.log( myList.serialize() );
+  ```
+  @codepen
 
 @body
 
 ## Use
 
 ```js
-import { DefineMap } from "can";
+import {DefineMap} from "can";
 
 const Address = DefineMap.extend( {
-	street: "string",
-	city: "string"
+    street: "string",
+    city: {type:"string", default: "Chicago"}
 } );
 
 const Direction = DefineMap.extend( {
-	from: { Type: Address },
-	to: Address
+    from: { Type: Address },
+    to: Address
 } );
 
 const direction = new Direction( {
-	from: { street: "2060 N. Stave", city: "Chicago" },
-	to: new Address( { street: "123 Greenview", city: "Libertyville" } )
+    from: {street: "2060 N. Stave"},
+    to: new Address( {street: "123 Greenview", city: "Libertyville"} )
 } );
+
+console.log( direction.serialize() ); //-> {
```

Comment:
These are addresses.  We should make sure we show this.  One way might be to write out:

```
console.log( direction.from )
```

Or maybe:

```
console.log( canReflect.getName(direction.from ) ) //-> Address{}
```

Reply:
It's worth noting: `direction.from` outputs the entire constructor, the latter outputs `"DefineMap{}"`.

I opted on `console.log( direction.from instanceof Address ); //-> true`